### DISCUSSION TITLE
OAuth2PasswordRequestForm   always error: 'There was an error parsing the body'

### DISCUSSION THREAD

**Discussion by cnly1987** [fastapi/fastapi#7491] [Questions] [answered]
if i use OAuth2PasswordRequestForm,  then  response : "detail": "There was an error parsing the body"

but i use body  everything is ok!  why?

`
@router.post("/token", response_model=schemas.Token)
async def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends(), db:Session = Depends(get_db)):
# async def login_for_access_token(*, username: str = Body(...), password: str = Body(...), db:Session = Depends(get_db)):
    user = authenticate_user(db, form_data.username, form_data.password)
    if not user:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Incorrect username or password",
            headers={"WWW-Authenticate": "Bearer"},
        )
    access_token_expires = timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
    access_token = create_access_token(
        data={"sub": user.username}, expires_delta=access_token_expires
    )
    return {"access_token": access_token, "token_type": "bearer"}
`

**Comment by tiangolo** [comment]
I assume you solved it, thanks for closing the issue :+1:

**Comment by marcelombc** [comment, marked as answer]
If you want to use the OAuth2PasswordRequestForm you need to install python-multipart package because OAuth2 uses "form data" for sending the username and password

**Comment by tom-on-the-internet** [comment]
Just in case you are like me. After you install `python-multipart`, remember to rebuild your docker container. Ha ha.
